### PR TITLE
feat: meta data and favicon

### PR DESCRIPTION
**Pull request by amitojsingh**

### CODE REVIEW

**Comment by MySmilingTurban** [comment]
@amitojsingh veerji, would it be possible to make the favicon a little bigger?
maybe remove the white border and make the letter bigger so it is readable.

File: src/pages/word/semantics/index.tsx [hunk 1]
```diff
@@ -157,39 +162,40 @@ export default function Semantics() {
...
     processWords(antonyms, 'antonyms');
-  }, [synonyms, antonyms, words, currentWord]);  
-  
+  }, [synonyms, antonyms, words, currentWord]);
 
   if (!currentWord.word) {
     // Handle case when word is not found
     return <div>{text('WORD_NOT_FOUND')}</div>;
   }
+  const { title, description } = metaTags.SEMANTICS;
```
Comment (MySmilingTurban) [inline src/pages/word/semantics/index.tsx:174]: move const definition before return
ideally any definition or functions should be before any return statement (got into a similar warning last week)
simply move it above the following code
```
if (!currentWord.word) {
    // Handle case when word is not found
    return <div>{text('WORD_NOT_FOUND')}</div>;
}
```

File: src/pages/word/information/index.tsx [hunk 2]
```diff
@@ -55,93 +66,127 @@ export default function Information() {
     // Handle case when word is not found
     return <div>{text('WORD_NOT_FOUND')}</div>;
   }
+  const { title, description } = metaTags.INFORMATION;
```
Comment (MySmilingTurban) [inline src/pages/word/information/index.tsx:69]: move const definition before conditional return statement

File: src/pages/word/information/index.tsx [hunk 3]
```diff
@@ -55,93 +66,127 @@ export default function Information() {
     // Handle case when word is not found
     return <div>{text('WORD_NOT_FOUND')}</div>;
```
Comment (MySmilingTurban) [inline src/pages/word/information/index.tsx:67]: Should we add a redirect to the "word not found" page (needs to be created) instead?
would also need metadata for that specific page.
please create an issue for this if needed

File: src/pages/word/layout.tsx [hunk 4]
```diff
@@ -1,10 +1,14 @@
+import Meta from 'components/meta';
+import metaTags from 'constants/meta';
 import React from 'react';
 import { Outlet } from 'react-router-dom';
 
 export default function WordsPageLayout() {
-  // check if children contains 
+  // check if children contains
+  const { title, description } = metaTags.WORD;
```
Comment (MySmilingTurban) [inline src/pages/word/layout.tsx:8]: This metadata will be overwritten by the respective pages (word/definition, examples, semantics, information), as this acts as a common layout for them.

**Comment by amitojsingh** [comment]
Sure veer ji I can try

File: src/pages/word/examples/index.tsx [hunk 5]
```diff
@@ -6,12 +6,14 @@ import { useLocation } from 'react-router-dom';
...
 import { useTranslation } from 'react-i18next';
+import Meta from 'components/meta';
+import metaTags from 'constants/meta';
 
 export default function Examples() {
   const { t: text } = useTranslation();
   // Use useLocation to get the search parameters from the URL
   const location = useLocation();
-
+  const { title, description } = metaTags.EXAMPLES;
```
Comment (MySmilingTurban) [inline src/pages/word/examples/index.tsx:16]: best position!
can we move all these metadata definitions to this position in every file, like 2-3 lines after function definition?
(to make code more readable)

File: src/pages/questions/index.tsx [hunk 6]
```diff
@@ -49,13 +61,28 @@ export default function Question() {
     return <div>{text('WORD_NOT_FOUND')}</div>;
   }
 
+  const { title, description } = metaTags.QUESTION;
```
Comment (MySmilingTurban) [inline src/pages/questions/index.tsx:64]: refer to metadata comment and change accordingly

File: src/pages/layout.tsx [hunk 7]
```diff
@@ -2,17 +2,14 @@ import React from 'react';
...
 import { Outlet } from 'react-router-dom';
-import { useTranslation } from 'react-i18next';
+import Meta from 'components/meta';
+import metaTags from 'constants/meta';
 
 export default function RootLayout() {
-  const { t: text } = useTranslation();
+  const { title, description } = metaTags.ROOT;
   return (
     <html lang='en'>
```
Comment (MySmilingTurban) [inline src/pages/layout.tsx:11]: do we still need html tag?

**Review by MySmilingTurban — CHANGES_REQUESTED** [review]

File: src/pages/word/information/index.tsx (hunk 3 above)
Comment (amitojsingh) [reply src/pages/word/information/index.tsx:67]: I think for now we can redirect to 404 not found. what do you think?

File: src/pages/word/semantics/index.tsx (hunk 1 above)
Comment (amitojsingh) [reply src/pages/word/semantics/index.tsx]: fixed veer ji

File: src/pages/word/examples/index.tsx (hunk 5 above)
Comment (amitojsingh) [reply src/pages/word/examples/index.tsx:16]: okay veer ji

File: src/pages/layout.tsx (hunk 7 above)
Comment (amitojsingh) [reply src/pages/layout.tsx]: no I removed it

File: src/pages/questions/index.tsx (hunk 6 above)
Comment (amitojsingh) [reply src/pages/questions/index.tsx]: done veer ji

File: src/pages/word/layout.tsx (hunk 4 above)
Comment (amitojsingh) [reply src/pages/word/layout.tsx]: removed veer ji

File: src/pages/word/information/index.tsx (hunk 2 above)
Comment (amitojsingh) [reply src/pages/word/information/index.tsx]: done veer ji

**Comment by amitojsingh** [comment]
@MySmilingTurban Veer ji I have updated the favicon.

File: src/pages/word/information/index.tsx (hunk 3 above)
Comment (MySmilingTurban) [reply src/pages/word/information/index.tsx]: ya, sounds good

**Review by MySmilingTurban — APPROVED** [review]
looks good to me!

**Pull request merged**